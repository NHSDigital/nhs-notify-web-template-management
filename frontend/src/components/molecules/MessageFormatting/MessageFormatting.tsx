import { TemplateType } from 'nhs-notify-web-template-management-utils';
import content from '@content/content';
import {
  BoldText,
  BulletList,
  Headings,
  HorizontalRule,
  LineBreaksAndParagraphs,
  LinksAndUrlsMarkdown,
  LinksAndUrlsNoMarkdown,
  NumberedList,
} from './formats';

const messageFormattingContent = content.components.messageFormattingComponent;

const messageFormattingMap: Record<TemplateType, JSX.Element[]> = {
  [TemplateType.NHS_APP]: [
    LineBreaksAndParagraphs(),
    Headings(),
    BoldText(),
    LinksAndUrlsMarkdown(),
  ],
  [TemplateType.EMAIL]: [
    LineBreaksAndParagraphs(),
    Headings(),
    BulletList(),
    NumberedList(),
    HorizontalRule(),
    LinksAndUrlsMarkdown(),
  ],
  [TemplateType.SMS]: [LinksAndUrlsNoMarkdown()],
};

export function MessageFormatting({ template }: { template: TemplateType }) {
  return (
    <>
      <h2
        className='nhsuk-heading-m nhsuk-u-margin-top-4'
        data-testid='personalisation-header'
      >
        {messageFormattingContent.header}
      </h2>
      {...messageFormattingMap[template]}
    </>
  );
}
