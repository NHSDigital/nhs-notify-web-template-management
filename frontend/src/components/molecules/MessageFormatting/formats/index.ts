export * from './BoldText';
export * from './BulletList';
export * from './Headings';
export * from './HorizontalRule';
export * from './LineBreaksAndParagraphs';
export * from './LinksAndUrls';
export * from './NumberedList';
