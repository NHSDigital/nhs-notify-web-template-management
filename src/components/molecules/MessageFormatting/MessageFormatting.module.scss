.inlineText {
  display: block;
}

.codeLine {
  > p:last-child {
      margin-bottom: 0;
  }
}

.horizontalLine {
  > p:not(:first-child){
      margin-bottom: 0;
    }
}
