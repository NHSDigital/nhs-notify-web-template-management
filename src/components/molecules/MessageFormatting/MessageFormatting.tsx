import styles from './MessageFormatting.module.scss';
import { Details } from 'nhsuk-react-components';
import { MessageFormattingType } from './message-formatting.types';
import { TemplateFormatText } from '@/src/types/template-format.types';

import content from '@/src/content/content';

const messageFormattingContent = content.components.messageFormattingComponent;

export function MessageFormatting({ template }: MessageFormattingType) {
  return (
    <>
      <h2
        className='nhsuk-heading-m nhsuk-u-margin-top-4'
        data-testid='personalisation-header'
      >
        {template === TemplateFormatText.LETTER
          ? messageFormattingContent.letterHeader
          : messageFormattingContent.header}
      </h2>

      <Details data-testid='lines-breaks-and-paragraphs-details'>
        <Details.Summary data-testid='lines-breaks-and-paragraphs-summary'>
          {messageFormattingContent.lineBreaksAndParagraphs.title}
        </Details.Summary>
        <Details.Text data-testid='lines-breaks-and-paragraphs-text'>
          <p>{messageFormattingContent.lineBreaksAndParagraphs.text1}</p>
          <code>
            {messageFormattingContent.lineBreaksAndParagraphs.codeBlockText.map(
              ({ id, item }) => (
                <span className={styles.inlineText} key={id}>
                  {item}
                </span>
              )
            )}
          </code>
          <p className='nhsuk-u-margin-top-4'>
            {messageFormattingContent.lineBreaksAndParagraphs.text2}
          </p>
          <code className={styles.codeLine}>
            {messageFormattingContent.lineBreaksAndParagraphs.codeBlockText.map(
              ({ id, item }) => (
                <p key={id}>{item}</p>
              )
            )}
          </code>
        </Details.Text>
      </Details>
      <Details data-testid='link-and-url-details'>
        <Details.Summary data-testid='link-and-url-summary'>
          {messageFormattingContent.linksAndUrls.title}
        </Details.Summary>
        <Details.Text data-testid='link-and-url-text'>
          <p>{messageFormattingContent.linksAndUrls.text1}</p>
          <p>{messageFormattingContent.linksAndUrls.text2}</p>
          <code>
            {messageFormattingContent.linksAndUrls.codeBlockText.text1}
          </code>
          <p className='nhsuk-u-margin-top-4'>
            {messageFormattingContent.linksAndUrls.text3}
          </p>
          <p>{messageFormattingContent.linksAndUrls.text4}</p>
          <code>
            {messageFormattingContent.linksAndUrls.codeBlockText.text2}
          </code>
        </Details.Text>
      </Details>
      {template !== TemplateFormatText.SMS ? (
        <Details data-testid='bold-text-details'>
          <Details.Summary data-testid='bold-text-summary'>
            {messageFormattingContent.boldText.title}
          </Details.Summary>
          <Details.Text data-testid='bold-text-text'>
            <p>{messageFormattingContent.boldText.text}</p>
            <code>{messageFormattingContent.boldText.codeBlockText}</code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.EMAIL ||
      template === TemplateFormatText.LETTER ? (
        <Details data-testid='headings-details'>
          <Details.Summary data-testid='headings-summary'>
            {messageFormattingContent.headings.title}
          </Details.Summary>
          <Details.Text data-testid='headings-text'>
            <p>{messageFormattingContent.headings.text1}</p>
            <code>{messageFormattingContent.headings.codeBlock.text1}</code>
            <p>{messageFormattingContent.headings.text2}</p>
            <code>{messageFormattingContent.headings.codeBlock.text2}</code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.EMAIL ||
      template === TemplateFormatText.LETTER ? (
        <Details data-testid='bullet-lists-details'>
          <Details.Summary data-testid='bullet-lists-summary'>
            {messageFormattingContent.bulletLists.title}
          </Details.Summary>
          <Details.Text data-testid='bullet-lists-text'>
            <p>{messageFormattingContent.bulletLists.text}</p>
            <code>
              {messageFormattingContent.bulletLists.codeBlockText.map(
                ({ id, item }) => (
                  <span className={styles.inlineText} key={id}>
                    {item}
                  </span>
                )
              )}
            </code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.EMAIL ||
      template === TemplateFormatText.LETTER ? (
        <Details data-testid='numbered-list-details'>
          <Details.Summary data-testid='numbered-list-summary'>
            {messageFormattingContent.numberedLists.title}
          </Details.Summary>
          <Details.Text data-testid='numbered-list-text'>
            <p>{messageFormattingContent.numberedLists.text}</p>
            <code>
              {messageFormattingContent.numberedLists.codeBlockText.map(
                ({ id, item }) => (
                  <span className={styles.inlineText} key={id}>
                    {item}
                  </span>
                )
              )}
            </code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.EMAIL ||
      template === TemplateFormatText.LETTER ? (
        <Details data-testid='signatures-details'>
          <Details.Summary data-testid='signatures-summary'>
            {messageFormattingContent.signatures.title}
          </Details.Summary>
          <Details.Text data-testid='signatures-text'>
            <p>{messageFormattingContent.signatures.text}</p>
            <code>{messageFormattingContent.signatures.codeBlockText}</code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.EMAIL ||
      template === TemplateFormatText.LETTER ? (
        <Details data-testid='horizontal-lines-details'>
          <Details.Summary data-testid='horizontal-lines-summary'>
            {messageFormattingContent.horizontalLine.title}
          </Details.Summary>
          <Details.Text data-testid='horizontal-lines-text'>
            <p>{messageFormattingContent.horizontalLine.text}</p>
            <code className={styles.horizontalLine}>
              {messageFormattingContent.horizontalLine.codeBlockText.map(
                ({ id, item }) => (
                  <p key={id}>{item}</p>
                )
              )}
            </code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.LETTER ? (
        <Details data-testid='page-breaks-details'>
          <Details.Summary data-testid='page-breaks-summary'>
            {messageFormattingContent.pageBreaks.title}
          </Details.Summary>
          <Details.Text data-testid='page-breaks-text'>
            <p>{messageFormattingContent.pageBreaks.text}</p>
            <code>
              {messageFormattingContent.pageBreaks.codeBlockText.map(
                ({ id, item }) => (
                  <span className={styles.inlineText} key={id}>
                    {item}
                  </span>
                )
              )}
            </code>
          </Details.Text>
        </Details>
      ) : null}
      {template === TemplateFormatText.LETTER ? (
        <Details data-testid='content-blocks-details'>
          <Details.Summary data-testid='content-blocks-summary'>
            {messageFormattingContent.contentBlocks.title}
          </Details.Summary>
          <Details.Text data-testid='content-blocks-text'>
            <p>{messageFormattingContent.contentBlocks.text1}</p>
            <p>{messageFormattingContent.contentBlocks.text2}</p>
            <p>{messageFormattingContent.contentBlocks.text3}</p>
            <code>
              {messageFormattingContent.contentBlocks.codeBlockText.map(
                ({ id, item }) => (
                  <span className={styles.inlineText} key={id}>
                    {item}
                  </span>
                )
              )}
            </code>
          </Details.Text>
        </Details>
      ) : null}
    </>
  );
}
