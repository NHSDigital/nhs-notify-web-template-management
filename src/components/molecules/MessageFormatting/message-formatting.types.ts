import { TemplateFormatText } from '@/src/types/template-format.types';

export type MessageFormattingType = {
  template: keyof typeof TemplateFormatText;
};
