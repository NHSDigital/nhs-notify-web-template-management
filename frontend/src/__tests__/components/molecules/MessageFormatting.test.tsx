import { render, screen } from '@testing-library/react';
import { MessageFormatting } from '@molecules/MessageFormatting/MessageFormatting';

const componentProps = {
  template: 'SMS',
};

describe('MessageFormatting component', () => {
  it('renders component correctly with SMS related formatting', () => {
    render(<MessageFormatting {...componentProps} />);

    expect(screen.getByTestId('personalisation-header')).toHaveTextContent(
      'Message formatting'
    );
    expect(screen.getByTestId('link-and-url-details')).toBeInTheDocument();
  });

  it('renders component correctly with APP related formatting', () => {
    const appFormattingProps = {
      ...componentProps,
      template: 'NHS_APP',
    };
    render(<MessageFormatting {...appFormattingProps} />);

    expect(screen.getByTestId('personalisation-header')).toHaveTextContent(
      'Message formatting'
    );
    expect(
      screen.getByTestId('lines-breaks-and-paragraphs-details')
    ).toBeInTheDocument();
    expect(screen.getByTestId('headings-details')).toBeInTheDocument();
    expect(screen.getByTestId('bold-text-details')).toBeInTheDocument();
    expect(screen.getByTestId('link-and-url-details')).toBeInTheDocument();
  });

  it('renders component correctly with EMAIL related formatting', () => {
    const emailFormattingProps = {
      ...componentProps,
      template: 'EMAIL',
    };
    render(<MessageFormatting {...emailFormattingProps} />);

    expect(screen.getByTestId('personalisation-header')).toHaveTextContent(
      'Message formatting'
    );
    expect(
      screen.getByTestId('lines-breaks-and-paragraphs-details')
    ).toBeInTheDocument();
    expect(screen.getByTestId('headings-details')).toBeInTheDocument();
    expect(screen.getByTestId('bullet-lists-details')).toBeInTheDocument();
    expect(screen.getByTestId('numbered-list-details')).toBeInTheDocument();
    expect(screen.getByTestId('horizontal-lines-details')).toBeInTheDocument();
    expect(screen.getByTestId('link-and-url-details')).toBeInTheDocument();
  });
});
