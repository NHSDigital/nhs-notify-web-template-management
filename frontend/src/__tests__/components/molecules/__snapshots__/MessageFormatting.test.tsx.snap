// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageFormatting component matches snapshot for EMAIL 1`] = `
<div>
  <h2
    class="nhsuk-heading-m nhsuk-u-margin-top-4"
    data-testid="message-formatting-header"
  >
    Message formatting
  </h2>
  <details
    class="nhsuk-details"
    data-testid="line-breaks-and-paragraphs-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="line-breaks-and-paragraphs-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Line breaks and paragraphs
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="line-breaks-and-paragraphs-text"
    >
      <p>
        To add a line break, use 2 spaces at the end of your text.
      </p>
      <p>
        Copy this example to add line breaks:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="linebreaks-markdown-description"
      >
        An example of line break markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="linebreaks-markdown-description"
        >
          line 1  
line 2  
line 3  
        </code>
      </pre>
      <p>
        To add a paragraph, use a blank line between each paragraph.
      </p>
      <p>
        Copy this example to add paragraphs:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="paragraphs-markdown-description"
      >
        An example of paragraph markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="paragraphs-markdown-description"
        >
          line 1

line 2

line 3
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="headings-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="headings-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Headings
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="headings-text"
    >
      <p>
        Use one hash symbol followed by a space for a heading.
      </p>
      <p>
        Copy this example to add a heading:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="headings-markdown-description"
      >
        An example of heading markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="headings-markdown-description"
        >
          # This is a heading
        </code>
      </pre>
      <p>
        To add a subheading, use 2 hash symbols:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="subheadings-markdown-description"
      >
        An example of subheading markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="subheadings-markdown-description"
        >
          ## This is a subheading
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="bullet-points-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="bullet-points-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Bullet points
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="bullet-points-text"
    >
      <p>
        Put each item on a separate line with an asterisk and a space in front of each one.
      </p>
      <p>
        Leave an empty line before the first bullet point and after the last bullet point.
      </p>
      <p>
        Copy this example to add bullet points:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="bullet-points-markdown-description"
      >
        An example of bullet point markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="bullet-points-markdown-description"
        >
          * bullet 1
* bullet 2
* bullet 3
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="numbered-lists-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="numbered-lists-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Numbered lists
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="numbered-lists-text"
    >
      <p>
        Put each item on a separate line with the number, full stop and a space in front of each one.
      </p>
      <p>
        Leave an empty line before the first item and after the last item.
      </p>
      <p>
        Copy this example to add a numbered list:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="numbered-list-markdown-description"
      >
        An example of numbered list markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="numbered-list-markdown-description"
        >
          1. first item
2. second item
3. third item
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="horizontal-lines-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="horizontal-lines-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Horizontal lines
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="horizontal-lines-text"
    >
      <p>
        To add a horizontal line between 2 paragraphs, use 3 dashes. Leave one empty line space after the first paragraph.
      </p>
      <p>
        Copy this example to add a horizontal line:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="horizontal-line-markdown-description"
      >
        An example of horizontal line markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="horizontal-line-markdown-description"
        >
          First paragraph

---
Second paragraph
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="links-and-urls-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="links-and-urls-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Links and URLs
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="links-and-urls-text"
    >
      <p>
        To convert text into a link, use square brackets around the link text and round brackets around the full URL. Make sure there are no spaces between the brackets or the link will not work.
      </p>
      <p>
        Copy this example to add a link:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="text-links-markdown-description"
      >
        An example of link markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="text-links-markdown-description"
        >
          [Read more](https://www.nhs.uk/)
        </code>
      </pre>
      <p>
        If you want to include the URL in full, use square brackets around the full URL to make it the link text and use round brackets around the full URL.
      </p>
      <p>
        Copy this example to add a URL:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="full-urls-markdown-description"
      >
        An example of URL markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="full-urls-markdown-description"
        >
          [https://www.nhs.uk/](https://www.nhs.uk/)
        </code>
      </pre>
    </div>
  </details>
</div>
`;

exports[`MessageFormatting component matches snapshot for LETTER 1`] = `
<div>
  <h2
    class="nhsuk-heading-m nhsuk-u-margin-top-4"
    data-testid="message-formatting-header"
  >
    Message formatting
  </h2>
</div>
`;

exports[`MessageFormatting component matches snapshot for NHS_APP 1`] = `
<div>
  <h2
    class="nhsuk-heading-m nhsuk-u-margin-top-4"
    data-testid="message-formatting-header"
  >
    Message formatting
  </h2>
  <details
    class="nhsuk-details"
    data-testid="line-breaks-and-paragraphs-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="line-breaks-and-paragraphs-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Line breaks and paragraphs
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="line-breaks-and-paragraphs-text"
    >
      <p>
        To add a line break, use 2 spaces at the end of your text.
      </p>
      <p>
        Copy this example to add line breaks:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="linebreaks-markdown-description"
      >
        An example of line break markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="linebreaks-markdown-description"
        >
          line 1  
line 2  
line 3  
        </code>
      </pre>
      <p>
        To add a paragraph, use a blank line between each paragraph.
      </p>
      <p>
        Copy this example to add paragraphs:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="paragraphs-markdown-description"
      >
        An example of paragraph markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="paragraphs-markdown-description"
        >
          line 1

line 2

line 3
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="headings-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="headings-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Headings
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="headings-text"
    >
      <p>
        Use one hash symbol followed by a space for a heading.
      </p>
      <p>
        Copy this example to add a heading:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="headings-markdown-description"
      >
        An example of heading markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="headings-markdown-description"
        >
          # This is a heading
        </code>
      </pre>
      <p>
        To add a subheading, use 2 hash symbols:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="subheadings-markdown-description"
      >
        An example of subheading markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="subheadings-markdown-description"
        >
          ## This is a subheading
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="bold-text-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="bold-text-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Bold text
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="bold-text-text"
    >
      <p>
        Use two asterisk symbols on either side of the words you want to be bold.
      </p>
      <p>
        Copy this example to add bold text:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="bold-text-markdown-description"
      >
        An example of bold text markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="bold-text-markdown-description"
        >
          **this is bold text**
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="bullet-points-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="bullet-points-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Bullet points
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="bullet-points-text"
    >
      <p>
        Put each item on a separate line with an asterisk and a space in front of each one.
      </p>
      <p>
        Leave an empty line before the first bullet point and after the last bullet point.
      </p>
      <p>
        Copy this example to add bullet points:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="bullet-points-markdown-description"
      >
        An example of bullet point markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="bullet-points-markdown-description"
        >
          * bullet 1
* bullet 2
* bullet 3
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="numbered-lists-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="numbered-lists-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Numbered lists
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="numbered-lists-text"
    >
      <p>
        Put each item on a separate line with the number, full stop and a space in front of each one.
      </p>
      <p>
        Leave an empty line before the first item and after the last item.
      </p>
      <p>
        Copy this example to add a numbered list:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="numbered-list-markdown-description"
      >
        An example of numbered list markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="numbered-list-markdown-description"
        >
          1. first item
2. second item
3. third item
        </code>
      </pre>
    </div>
  </details>
  <details
    class="nhsuk-details"
    data-testid="links-and-urls-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="links-and-urls-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Links and URLs
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="links-and-urls-text"
    >
      <p>
        To convert text into a link, use square brackets around the link text and round brackets around the full URL. Make sure there are no spaces between the brackets or the link will not work.
      </p>
      <p>
        Copy this example to add a link:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="text-links-markdown-description"
      >
        An example of link markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="text-links-markdown-description"
        >
          [Read more](https://www.nhs.uk/)
        </code>
      </pre>
      <p>
        If you want to include the URL in full, use square brackets around the full URL to make it the link text and use round brackets around the full URL.
      </p>
      <p>
        Copy this example to add a URL:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="full-urls-markdown-description"
      >
        An example of URL markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="full-urls-markdown-description"
        >
          [https://www.nhs.uk/](https://www.nhs.uk/)
        </code>
      </pre>
    </div>
  </details>
</div>
`;

exports[`MessageFormatting component matches snapshot for SMS 1`] = `
<div>
  <h2
    class="nhsuk-heading-m nhsuk-u-margin-top-4"
    data-testid="message-formatting-header"
  >
    Message formatting
  </h2>
  <details
    class="nhsuk-details"
    data-testid="links-and-urls-details"
  >
    <summary
      class="nhsuk-details__summary"
      data-testid="links-and-urls-summary"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Links and URLs
      </span>
    </summary>
    <div
      class="nhsuk-details__text"
      data-testid="links-and-urls-text"
    >
      <p>
        Write the URL in full, starting with https://
      </p>
      <p>
        For example:
      </p>
      <span
        class="nhsuk-u-visually-hidden"
        id="links-urls-markdown-description"
      >
        An example of URL markdown
      </span>
      <pre
        class="nhsuk-code-block"
      >
        <code
          aria-describedby="links-urls-markdown-description"
        >
          https://www.nhs.uk/example
        </code>
      </pre>
    </div>
  </details>
</div>
`;
